### PR TITLE
add documentation for filing package object

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- Adding details to the generate Url payload documentation

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

N/A

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,2 +1,204 @@
 # jag-file-submission
+
 Generic File Submission API (to be used by the Family Law Act Application at first)
+
+## Introduction
+
+Ready for integrating your application with bcgov efiling service and allow your users to efile document ?
```
Comment (sdevalapurkar) [inline docs/README.md:7]:
```suggestion
Ready to integrate your application with bcgov's eFiling service and allow your users to eFile documents?
```

File: docs/README.md [hunk 2]
```diff
@@ -1,2 +1,204 @@
...
+
+This guide will provide you with all information you need to get started.
```
Comment (sdevalapurkar) [inline docs/README.md:9]:
```suggestion
This guide will provide you with all the required information you need to get started.
```

File: docs/README.md (hunk 1 above)
Comment (akroon3r) [inline docs/README.md:7]:
```suggestion
Are you ready to integrate your application with the BC Gov E-filing Hub in order to allow your users to e-file documents?
```

File: docs/README.md [hunk 3]
```diff
@@ -1,2 +1,204 @@
...
+
+### Request a profile
+
+> Follow the following instructions in [Getting started - Onboarding](onboarding.md)
+
+### Start Integrating
+
+Get an oauth token from our Keycloak Server using the following curl
```
Comment (sdevalapurkar) [inline docs/README.md:17]:
```suggestion
Get an oauth token from our Keycloak Server using the following curl command:
```

File: docs/README.md (hunk 1 above)
Comment (akroon3r) [reply docs/README.md:7]: iunno tho

File: docs/README.md [hunk 4]
```diff
@@ -1,2 +1,204 @@
...
+curl --location --request POST 'url' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--data-urlencode 'client_id=[your client id]' \
+--data-urlencode 'grant_type=client_credentials' \
+--data-urlencode 'client_secret=[your client secret]'
+```
+
+> the secret should be kept safe and not exposed to the public
+
+Start by uploading document(s) to the **efiling hub** api
```
Comment (sdevalapurkar) [inline docs/README.md:29]:
```suggestion
Start by uploading document(s) to the **eFiling Hub** API:
```

File: docs/README.md [hunk 5]
```diff
@@ -1,2 +1,204 @@
...
+the response includes a submission id that you will use to generate a redirect url
+
+```json
+{
+  "submissionId": "5e9492cf-e87e-48b5-ba55-0c198d8edde5",
+  "received": 1
+}
+```
+
+Then generate a unique url to redirect the users to the **efiling hub**
```
Comment (sdevalapurkar) [inline docs/README.md:48]:
```suggestion
Then generate a unique url to redirect the users to the **eFiling Hub**
```

File: docs/README.md [hunk 6]
```diff
@@ -1,2 +1,204 @@
...
+### Request a profile
+
+> Follow the following instructions in [Getting started - Onboarding](onboarding.md)
+
+### Start Integrating
+
+Get an oauth token from our Keycloak Server using the following curl
+
+```bash
+curl --location --request POST 'url' \
```
Comment (akroon3r) [inline docs/README.md:20]:
```suggestion
curl --location --request POST 'keycloak_url' \
```

File: docs/README.md [hunk 7]
```diff
@@ -1,2 +1,204 @@
...
+
+> the secret should be kept safe and not exposed to the public
+
+Start by uploading document(s) to the **efiling hub** api
+
+```bash
+curl --location --request POST 'http://fla-nginx-proxy-qzaydf-dev.pathfinder.gov.bc.ca/api/submission/documents' \
+--header 'X-Transaction-Id: ca09e538-d34e-11ea-87d0-0242ac130003' \
+--header 'Content-Type: multipart/form-data' \
+--header 'Authorization: Bearer [your bearer token]' \
```
Comment (akroon3r) [inline docs/README.md:35]:
```suggestion
--header 'Authorization: Bearer [bearer_token]' \
```

File: docs/README.md [hunk 8]
```diff
@@ -1,2 +1,204 @@
...
+}
+```
+
+### Generate URL Payload Details
+
+#### navigation
+
+The `navigation` object represents a list of possible returns to your application based on the status of the document e-filing.
+
+#### clientAppplication
```
Comment (sdevalapurkar) [inline docs/README.md:105]:
```suggestion
#### clientApplication
```

File: docs/README.md [hunk 9]
```diff
@@ -1,2 +1,204 @@
...
+
+The `clientAppplication` object represents how your application is labelled in **efiling hub**.
```
Comment (sdevalapurkar) [inline docs/README.md:107]:
```suggestion
The `clientApplication` object represents how your application is labelled in **efiling hub**.
```

File: docs/README.md [hunk 10]
```diff
@@ -1,2 +1,204 @@
...
+--data-urlencode 'grant_type=client_credentials' \
+--data-urlencode 'client_secret=[your client secret]'
+```
+
+> the secret should be kept safe and not exposed to the public
+
+Start by uploading document(s) to the **efiling hub** api
+
+```bash
+curl --location --request POST 'http://fla-nginx-proxy-qzaydf-dev.pathfinder.gov.bc.ca/api/submission/documents' \
```
Comment (akroon3r) [inline docs/README.md:32]: is our reverse proxy URL meant to be public?

File: docs/README.md [hunk 11]
```diff
@@ -1,2 +1,204 @@
...
+
+The `navigation` object represents a list of possible returns to your application based on the status of the document e-filing.
+
+#### clientAppplication
+
+The `clientAppplication` object represents how your application is labelled in **efiling hub**.
+
+#### filingPackage
+
+The `filingPackage` object represents the court information about the submited package along with the document(s) info.
```
Comment (sdevalapurkar) [inline docs/README.md:111]:
```suggestion
The `filingPackage` object represents the court information about the submitted package along with the document(s) info.
```

File: docs/README.md [hunk 12]
```diff
@@ -1,2 +1,204 @@
...
+}
+```
+
+Then generate a unique url to redirect the users to the **efiling hub**
+
+```bash
+curl --location --request POST 'http://fla-nginx-proxy-qzaydf-dev.pathfinder.gov.bc.ca/api/submission/5e9492cf-e87e-48b5-ba55-0c198d8edde5/generateUrl' \
+--header 'X-Transaction-Id: ca09e538-d34e-11ea-87d0-0242ac130003' \
+--header 'Content-Type: application/json' \
+--header 'Authorization: Bearer [your bearer token]' \
```
Comment (akroon3r) [inline docs/README.md:54]:
```suggestion
--header 'Authorization: Bearer [bearer_token]' \
```

File: docs/README.md [hunk 13]
```diff
@@ -1,2 +1,204 @@
...
+| ---------- | ------ | ----------------------------------------------------------------------------- |
+| location   | string | Court House where the package is submitted to                                 |
+| level      | string | Court level, can be `Supreme` of `Provincial`                                 |
+| courtClass | string | Court Classification see [Court Classification](data.md#Court-Classification) |
+| division   | string | R for Criminal, I for Civil see [Court Division](data.md#Court-Division)      |
+| fileNumber | string | The court file number                                                         |
+
+##### Documents
+
+The `documents` array represents the previously uploaded document.
```
Comment (sdevalapurkar) [inline docs/README.md:129]:
```suggestion
The `documents` array represents the previously uploaded documents.
```

File: docs/README.md [hunk 14]
```diff
@@ -1,2 +1,204 @@
...
+
+###### Properties
+
+| name                     | type    | description                                                                                |
+| ------------------------ | ------- | ------------------------------------------------------------------------------------------ |
+| name                     | string  | the document name, must be the same name as uploaded previously                            |
+| type                     | string  | the type of document see [Document Type Codes](data.md#Document-Type-Codes)                |
+| isAmendment              | boolean | if the document is an amendment                                                            |
+| isSupremeCourtScheduling | boolean | is the document is directed to Supreme Court Scheduling                                    |
```
Comment (sdevalapurkar) [inline docs/README.md:138]:
```suggestion
| isSupremeCourtScheduling | boolean | if the document is directed to Supreme Court Scheduling                                    |
```

File: docs/README.md [hunk 15]
```diff
@@ -1,2 +1,204 @@
...
+| data                     | object  | A non define json object representing the form data of the document                        |
```
Comment (sdevalapurkar) [inline docs/README.md:139]:
```suggestion
| data                     | object  | A non defined json object representing the form data of the document                        |
```

**Review by akroon3r — COMMENTED** [review]
Suggestions and one question.

File: docs/README.md [hunk 16]
```diff
@@ -1,2 +1,204 @@
...
+##### Properties
+
+| name       | type   | description                                                                                     |
+| ---------- | ------ | ----------------------------------------------------------------------------------------------- |
+| partyType  | string | the party type, IND for individual or ORG for Organization see [Party Type](data.md#Party Type) |
+| roleType   | string | the party role, see [Party Role](data.md#Party-Role)                                            |
+| firstName  | string | the party first name                                                                            |
+| middleName | string | the party middle name                                                                           |
+| lastName   | string | the party last name                                                                             |
+
```
Comment (sdevalapurkar) [inline docs/README.md:155]: as part of the generate url payload info we should also include a section about the validation.

1) must provide a court file number. If this is not provided, then parties with valid role type are mandatory

2) must provide valid document types

3) must provide a valid court level, class, and location combination

File: docs/README.md [hunk 17]
```diff
@@ -1,2 +1,204 @@
...
+### Api Documentation
+
+E-filing-api is documented using [openapi](http://editor.swagger.io/?url=https://raw.githubusercontent.com/bcgov/jag-file-submission/master/src/backend/efiling-api/jag-efiling-api.yaml) and we have a [postman collection](https://raw.githubusercontent.com/bcgov/jag-file-submission/master/src/backend/jag-efiling-api/src/test/jag-efiling-api.postman_collection.json) that you can use to test the e-filing-api
```
Comment (sdevalapurkar) [inline docs/README.md:158]:
```suggestion
eFiling API is documented using [openapi](http://editor.swagger.io/?url=https://raw.githubusercontent.com/bcgov/jag-file-submission/master/src/backend/efiling-api/jag-efiling-api.yaml) and we have a [postman collection](https://raw.githubusercontent.com/bcgov/jag-file-submission/master/src/backend/jag-efiling-api/src/test/jag-efiling-api.postman_collection.json) that you can use to test the eFiling API
```

**Review by sdevalapurkar — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 16 above)
Comment (alexjoybc) [reply docs/README.md:155]: I have added a required column in all the tables

File: docs/README.md (hunk 10 above)
Comment (alexjoybc) [reply docs/README.md]: removed

**Review by TayGov — APPROVED** [review]

**Pull request merged**